src/components/structures/RightPanel.js

```diff
@@ -66,15 +67,19 @@ module.exports = React.createClass({
     onRoomStateMember: function(ev, state, member) {
         // redraw the badge on the membership list
         if (this.state.phase == this.Phase.MemberList && member.roomId === this.props.roomId) {
-            this.forceUpdate();
+            this._delayedUpdate();
         }
         else if (this.state.phase === this.Phase.MemberInfo && member.roomId === this.props.roomId &&
                 member.userId === this.state.member.userId) {
             // refresh the member info (e.g. new power level)
-            this.forceUpdate();
+            this._delayedUpdate();
         }
     },
 
+    _delayedUpdate: new rate_limited_func(function() {
+        this.forceUpdate()
```

Comment:
Yeah, I made the rate_limited_func function so you could still use this as you expect, so it works out.